### PR TITLE
Separate Pirate Fortress ER Option

### PR DESCRIPTION
**Pull request by XenoWars**
This adds an option that separates the Pirate Fortress entrances from the relevant shuffles. The Sewers exit door will be shuffled if this and Extra Interiors are enabled, the Sewers gate and two interior entrances will be shuffled if this and Overworld ER are enabled, and if the previous is true then Pirate Fortress will no longer be considered a dungeon and shuffle the main entrance among the overworld as well.

### CODE REVIEW

File: packages/core/lib/combo/patch-build/randomizer.ts [hunk 1]
```diff
@@ -720,7 +720,7 @@ const gameEntrances = (worldId: number, game: Game, logic: LogicResult) => {
   return padBuffer16(toU32Buffer(data));
 };
 
-const randomizerDungeonsBits = (worldId: number, logic: LogicResult): Buffer => {
+const randomizerDungeonsBits = (worldId: number, logic: LogicResult, settings: Settings): Buffer => {
```
Comment (CelestialKitsune) [inline packages/core/lib/combo/patch-build/randomizer.ts:723]: No need to pass the `settings`, they're accessible through the `logic` object, please use `logic.settings` instead within the function

File: packages/core/lib/combo/patch-build/randomizer.ts [hunk 2]
```diff
@@ -761,6 +761,9 @@ const randomizerDungeonsBits = (worldId: number, logic: LogicResult): Buffer =>
 
   for (let i = 0; i < DUNGEONS_PRECOMPLETED.length; ++i) {
     const dungeon = DUNGEONS_PRECOMPLETED[i];
+    if (settings.erPiratesWorld && world.preCompleted.has('PF')) {
```
Comment (CelestialKitsune) [inline packages/core/lib/combo/patch-build/randomizer.ts:764]: This will never be true as PF cannot be a pre-completed dungeon.

**Review by CelestialKitsune — CHANGES_REQUESTED** [review]

File: packages/core/lib/combo/patch-build/randomizer.ts (hunk 2 above)
Comment (XenoWars) [reply packages/core/lib/combo/patch-build/randomizer.ts:764]: Well, PF is part of the pre-completed dungeon list, so I'm not quite sure what you mean by that.

Comment (CelestialKitsune) [reply packages/core/lib/combo/patch-build/randomizer.ts:764]: Check the [`selectPreCompletedDungeons`](https://github.com/OoTMM/OoTMM/blob/master/packages/core/lib/combo/logic/solve.ts#L810-L818) function and the other functions it calls. Only Major Dungeons can be pre-completed.

**Pull request merged**